### PR TITLE
Added row factory parameter to copy_rows function

### PR DESCRIPTION
**Pull request by Real-Gecko**

### CODE REVIEW

**Comment by volcan01010** [comment]
Thanks - this is really useful and we should have included before.

Can you add an integration test, please? It can be very similar to `test_copy_rows_happy_path`, but using the dict row factory.

https://github.com/BritishGeologicalSurvey/etlhelper/blob/5f1f1780589f5d4a8136dff223f95e7854512c6b/test/integration/etl/test_etl_transform.py#L12

There is information about setting up tests in CONTRIBUTORS.md.  The `test_etl_transform` function can be tested against a Dockerised PostgreSQL.

**Comment by Real-Gecko** [comment]
Added test.

**Comment by volcan01010** [comment]
The Travis tests failed on a Flake 8 error:

>$ flake8 etlhelper test
>test/integration/etl/test_etl_transform.py:102:13: E128 continuation line under-indented for visual indent
>The command "flake8 etlhelper test" exited with 1.

File: test/integration/etl/test_etl_transform.py [hunk 1]
```diff
@@ -84,3 +84,28 @@ def test_get_rows_with_modify_dict(pgtestdb_conn, pgtestdb_test_tables):
...
+    # Arrange
+    select_sql = "SELECT * FROM src"
+    insert_sql = """
+        INSERT INTO dest (id, value, simple_text, utf8_text, day, date_time)
+        VALUES (
+            %(id)s, %(value)s, %(simple_text)s,
+            %(utf8_text)s, %(day)s, %(date_time)s
+        );
+    """
+    expected = []
```
[the remark this answers is not in the document]
Comment (Real-Gecko) [reply test/integration/etl/test_etl_transform.py:99]: Ok

**Comment by Real-Gecko** [comment]
> The Travis tests failed on a Flake 8 error:
> 
> > $ flake8 etlhelper test
> > test/integration/etl/test_etl_transform.py:102:13: E128 continuation line under-indented for visual indent
> > The command "flake8 etlhelper test" exited with 1.

That's weird, I run `flake8 etlhelper test` on my laptop and do not get any errors.

**Comment by Real-Gecko** [comment]
> test/integration/etl/test_etl_transform.py:102:13

`test_etl_transform.py` does not have line 102, it ends on 87:
image

File: test/integration/etl/test_etl_transform.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Real-Gecko) [reply test/integration/etl/test_etl_transform.py]: Updated

**Comment by Real-Gecko** [comment]
I've updated the test, to have hardcoded values. Not sure about flake8 however, there's no errors on my environment. I guess build will show.

**Comment by Real-Gecko** [comment]
Ok, checks are passed now :D

**Comment by volcan01010** [comment]
This failed on the full test suite because `datetime` represented with class, not the module.  I fixed it and now I'm happy to merge.  Thanks for this - it really should have been there all along.

**Pull request merged**